File: app/src/main/java/me/imzack/app/ender/injector/component/AllTypesComponent.java
package me.imzack.app.ender.injector.component;

import me.imzack.app.ender.injector.module.AllTypesPresenterModule;
import me.imzack.app.ender.injector.scope.FragmentScope;
import me.imzack.app.ender.view.fragment.AllTypesFragment;

import dagger.Component;

@FragmentScope
@Component(dependencies = AppComponent.class, modules = AllTypesPresenterModule.class)
public interface AllTypesComponent {

    void inject(AllTypesFragment allTypesFragment);
}

File: app/src/main/java/net/zackzhang/app/end/injector/component/AllTypesComponent.kt
package net.zackzhang.app.end.injector.component

import dagger.Component
import net.zackzhang.app.end.injector.module.AllTypesPresenterModule
import net.zackzhang.app.end.injector.scope.FragmentScope
import net.zackzhang.app.end.view.fragment.AllTypesFragment

@FragmentScope
@Component(dependencies = arrayOf(AppComponent::class), modules = arrayOf(AllTypesPresenterModule::class))
interface AllTypesComponent {

    fun inject(allTypesFragment: AllTypesFragment)
}
